tfchain_bridge/pkg/bridge/bridge.go

```diff
@@ -102,6 +102,32 @@ func (bridge *Bridge) Start(ctx context.Context) error {
 			if data.Err != nil {
 				return errors.Wrap(err, "failed to process events")
 			}
+
+			// Handle pending withdraws / refunds that were not yet executed
+			if data.Head-uint64(height.LastHeight) >= bridge.config.RetryInterval {
+				pendingWithdraws, err := bridge.subClient.GetPendingWithdraws()
+				if err != nil {
+					return errors.Wrap(err, "failed to get pending withdraws")
+				}
+				for _, withdraw := range *pendingWithdraws {
+					err := bridge.handlePendingWithdraw(ctx, withdraw)
+					if err != nil {
+						return errors.Wrap(err, "failed to handle pending withdraw")
+					}
+				}
+
+				pendingRefunds, err := bridge.subClient.GetPendingRefunds()
+				if err != nil {
+					return errors.Wrap(err, "failed to get pending refunds")
+				}
+				for _, refund := range *pendingRefunds {
+					err := bridge.handlePendingRefund(ctx, refund)
+					if err != nil {
+						return errors.Wrap(err, "failed to handle pending refund")
+					}
+				}
+			}
```

Comment:
You also need to set the processed height in the persistency, otherwise the check at Line 107 wil always fire

Reply:
Not sure I get the persistency role...
Please check changes to check if expected logic is respected

Reply:
Well, you need to update the latest height in the persistency, now you are just overriding it with the same value.

If you don't persist it somewhere, the bridge will never know how to calculate the retry interval

Reply:
```
				err = bridge.blockPersistency.SaveHeight(data.Head)
				if err != nil {
					return errors.Wrap(err, "error while saving height")
				}
```

Reply:
I get it
Should be good now

I definitively need to read more about the bridge logic
Btw, how to test it?

Reply:
Some draft how to test it can be found here: https://github.com/threefoldtech/tfchain_tft_bridge/pull/113

Will prolly finish the docs tomorrow so we can do a full review and round of test